### ISSUE TITLE
[IOS 13] Application crashes on start of background geo location

### ISSUE THREAD

**Issue by jorendemaeyer1996** [transistorsoft/react-native-background-geolocation#1009]
## Your Environment
* Plugin version: 3.4.2
* Platform: iOS
* OS version: 13.3
* Device manufacturer / model: iPhone 11 
* React Native version (`react-native -v`): 0.61.5
* Plugin config
```javascript <!-- Syntax-highlighting:  paste your code below -->
    BackgroundGeolocation.ready(
      {
        debug: true, // <-- enable this hear debug sounds.
        logLevel: BackgroundGeolocation.LOG_LEVEL_VERBOSE,
        desiredAccuracy: BackgroundGeolocation.DESIRED_ACCURACY_HIGH,
        url: config.TRACKER_HOST + '/api/locations',
        stopOnTerminate: false, // <-- Allow the background-service to continue tracking when app terminated.
        startOnBoot: true,
        autoSync: true, // <-- Set true to sync each location to server as it arrives.
        foregroundService: true,
        preventSuspend: true,
        heartbeatInterval: 60,
        locationAuthorizationRequest: 'Always',
        disableLocationAuthorizationAlert: platform.isIos13() ? true : false,
        authorization: {
          strategy: 'JWT',
          accessToken: token.accessToken,
          refreshToken: token.refreshToken,
          refreshUrl: config.TRACKER_HOST + '/api/refresh_token',
          refreshPayload: {
            refresh_token: '{refreshToken}',
          },
          expires: token.expires,
        },
      },
      state => {
        console.log(
          '- BackgroundGeolocation is configured and ready: ',
          state.enabled,
        );

        this.setState({
          enabled: state.enabled,
        });
      },
    );
```

## Expected Behavior
Hi,

We are evaluating your sdk.
Currently we are testing it on react-native.
First of all it works quite smooth and well documented.

However we are still facing an issue on iOS 13. Once BackgroundGeolocation is ready and I want to call the start method the application closes without a warning/error.

Also regarding the background services we get very strange results. Suddenly the background sync stops and we only see fragmented data points.

Our goal is to limit battery use and to sync once every hour or in multiple batches. However it should sync all the historical data points which are now lost.

Could you help us its for a quite short deadline project. If this works we will buy the license directly.

## Actual Behavior
The application crashes/closes when I call the start method.
No errors are thrown or logged.

Also regarding the background services we get very strange results. Suddenly the background sync stops and we only see fragmented data points (see image).

<img width="624" alt="Screenshot 2020-03-18 at 18 44 46" src="https://user-images.githubusercontent.com/16215602/76991797-7a629980-694a-11ea-864b-676963e5daff.png">

## Steps to Reproduce

1. Setup react-native project
2. Setup is based on the example 'simple map' with the admin back-end.
3. When component is mounted set config
4. trigger the start on a button click or toggle.

## Context
See expected behaviour

## Debug logs

<details><summary>Logs</summary>

``` <!-- Syntax highlighting:  DO NOT REMOVE -->

```

</details>

**Comment by christocracy** [comment]
>The application crashes/closes when I call the start method.

Where's your stacktrace?

**Comment by jorendemaeyer1996** [comment]
`2020-03-18 20:38:39.347 [info][tid:com.facebook.react.JavaScript] stopBackgroundListener

╔═══════════════════════════════════════════════════════════
║ -[TSLocationManager start] 
╚═══════════════════════════════════════════════════════════
2020-03-18 20:38:39.352128+0100 CoronaTrackingApp[16111:509764] 
╔═══════════════════════════════════════════════════════════
║ -[TSLocationManager start] 
╚═══════════════════════════════════════════════════════════
ℹ️-[TSLocationManager doStart:] trackingMode: 1
2020-03-18 20:38:39.352354+0100 CoronaTrackingApp[16111:509763] ℹ️-[TSLocationManager doStart:] trackingMode: 1
ℹ️-[TSConfig persist] 
2020-03-18 20:38:39.352810+0100 CoronaTrackingApp[16111:510126] ℹ️-[TSConfig persist] 
🎾-[TSGeofenceManager start] 
2020-03-18 20:38:39.353664+0100 CoronaTrackingApp[16111:510126] 🎾-[TSGeofenceManager start] 
🔵-[TSLocationManager setPace:] 0
2020-03-18 20:38:39.354584+0100 CoronaTrackingApp[16111:509763] 🔵-[TSLocationManager setPace:] 0
🎾-[TSLocationManager startUpdatingLocation] Location-services: ON
2020-03-18 20:38:39.358745+0100 CoronaTrackingApp[16111:509764] 🎾-[TSLocationManager startUpdatingLocation] Location-services: ON

╔═══════════════════════════════════════════════════════════
║ -[TSHttpService flush:] 
╚═══════════════════════════════════════════════════════════
2020-03-18 20:38:39.362049+0100 CoronaTrackingApp[16111:509763] 
╔═══════════════════════════════════════════════════════════
║ -[TSHttpService flush:] 
╚═══════════════════════════════════════════════════════════

╔═══════════════════════════════════════════════════════════
║ -[TSHttpService finish:error:] Success: 1
╚═══════════════════════════════════════════════════════════
2020-03-18 20:38:39.362312+0100 CoronaTrackingApp[16111:509764] 
╔═══════════════════════════════════════════════════════════
║ -[TSHttpService finish:error:] Success: 1
╚═══════════════════════════════════════════════════════════
2020-03-18 20:38:39.369 [info][tid:com.facebook.react.JavaScript] '[event] location: ', { coords: 
   { altitude: 0,
     heading: 82.54,
     altitude_accuracy: -1,
     latitude: 37.33069273,
     floor: 0,
     accuracy: 10,
     longitude: -122.02971484,
     speed: 3.81 },
  timestamp: '2020-03-18T19:38:36.633Z',
  extras: {},
  sample: true,
  is_moving: false,
  uuid: '208F7BE2-4800-4F81-8D1A-BB0503519242',
  odometer: 1416.3,
  activity: { confidence: 100, type: 'unknown' },
  battery: { is_charging: false, level: -1 } }
ℹ️+[LocationAuthorization run:onCancel:] status: 3
2020-03-18 20:38:39.372361+0100 CoronaTrackingApp[16111:509763] ℹ️+[LocationAuthorization run:onCancel:] status: 3
🎾-[TSLocationManager startMonitoringBackgroundFetch] BackgroundFetch: ON
2020-03-18 20:38:39.373509+0100 CoronaTrackingApp[16111:510125] 🎾-[TSLocationManager startMonitoringBackgroundFetch] BackgroundFetch: ON
2020-03-18 20:38:39.373874+0100 CoronaTrackingApp[16111:507385] -[TSBackgroundFetch start]: unrecognized selector sent to instance 0x600003c880f0
2020-03-18 20:38:39.376352+0100 CoronaTrackingApp[16111:507385] *** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: '-[TSBackgroundFetch start]: unrecognized selector sent to instance 0x600003c880f0'
*** First throw call stack:
(
	0   CoreFoundation                      0x00007fff23c7127e __exceptionPreprocess + 350
	1   libobjc.A.dylib                     0x00007fff513fbb20 objc_exception_throw + 48
	2   CoreFoundation                      0x00007fff23c91fd4 -[NSObject(NSObject) doesNotRecognizeSelector:] + 132
	3   CoreFoundation                      0x00007fff23c75c4c ___forwarding___ + 1436
	4   CoreFoundation                      0x00007fff23c77f78 _CF_forwarding_prep_0 + 120
	5   CoronaTrackingApp                   0x000000010b4750eb -[TSLocationManager startMonitoringBackgroundFetch] + 345
	6   libdispatch.dylib                   0x000000010c5a3dd4 _dispatch_call_block_and_release + 12
	7   libdispatch.dylib                   0x000000010c5a4d48 _dispatch_client_callout + 8
	8   libdispatch.dylib                   0x000000010c5b2de6 _dispatch_main_queue_callback_4CF + 1500
	9   CoreFoundation                      0x00007fff23bd4049 __CFRUNLOOP_IS_SERVICING_THE_MAIN_DISPATCH_QUEUE__ + 9
	10  CoreFoundation                      0x00007fff23bceca9 __CFRunLoopRun + 2329
	11  CoreFoundation                      0x00007fff23bce066 CFRunLoopRunSpecific + 438
	12  GraphicsServices                    0x00007fff384c0bb0 GSEventRunModal + 65
	13  UIKitCore                           0x00007fff48092d4d UIApplicationMain + 1621
	14  CoronaTrackingApp                   0x000000010b05d710 main + 112
	15  libdyld.dylib                       0x00007fff5227ec25 start + 1
	16  ???                                 0x0000000000000001 0x0 + 1
)
libc++abi.dylib: terminating with uncaught exception of type NSException
(lldb) `

**Comment by christocracy** [comment]
Duplicate #993

**Comment by mujeebsardar** [comment]
@christocracy sir my application run on android fine but same code on IOS application will crash but after crashing on top bar library send the notification boot on background , location service on but application will crash i am facing this issues since last 2 days any suggestion i am. using a valid licence key purshase from the transistor shop

**Comment by christocracy** [comment]
This issue is closed and you're not providing a stacktrace.  There's no evidence that your problem has anything at all to do with this *closed* issue.

**Comment by mujeebsardar** [comment]
i will provide you kindly in contact

**Comment by mujeebsardar** [comment]
![simulator_screenshot_DEDA3E6F-7F55-4591-B6F4-CD2134ACD3A8](https://user-images.githubusercontent.com/80490730/112620704-b4625780-8e4a-11eb-8163-44381ecbb3d2.png)

**Comment by mujeebsardar** [comment]
@christocracy  recently i get this error i am using the same componentWillMount Function code as per provided in setup

**Comment by christocracy** [comment]
Your problem has nothing to do with this issues other than the word "*crash*".

Post a new issue.  Provide code.

**Comment by mujeebsardar** [comment]
i will genrate a new kindly give responce on it my deadline of project is near to complete kinldy reply 

https://github.com/transistorsoft/react-native-background-geolocation/issues/1273

**Comment by mujeebsardar** [comment]
nw_socket_handle_socket_event [C5.2:1] Socket SO_ERROR [61: Connection refused]
2021-03-26 16:19:24.533454+0500 TestApp[14012:243136] [connection] nw_connection_get_connected_socket [C5] Client called nw_connection_get_connected_socket on unconnected nw_connection
2021-03-26 16:19:24.538439+0500 TestApp[14012:243136] TCP Conn 0x60000187c370 Failed : error 0:61 [61]
2021-03-26 16:19:24.575365+0500 TestApp[14012:242742] Metal API Validation Enabled
2021-03-26 16:19:24.733083+0500 TestApp[14012:243106] ℹ️-[TSLocationManager on:success:failure:] location
2021-03-26 16:19:24.733308+0500 TestApp[14012:243106] ℹ️-[TSLocationManager on:success:failure:] motionchange
2021-03-26 16:19:24.736293+0500 TestApp[14012:243097] ℹ️-[TSLocationManager on:success:failure:] activitychange
2021-03-26 16:19:24.736910+0500 TestApp[14012:243097] ℹ️-[TSLocationManager on:success:failure:] providerchange
2021-03-26 16:19:24.829651+0500 TestApp[14012:243136] ℹ️-[TSConfig persist]
2021-03-26 16:19:24.833903+0500 TestApp[14012:243097] ℹ️-[TSConfig persist]
2021-03-26 16:19:24.851491+0500 TestApp[14012:243136] 🔵-[TSLocationManager ready]
2021-03-26 16:19:24.853112+0500 TestApp[14012:243106] ℹ️-[TSConfig didDeviceReboot] didDeviceReboot? 0
2021-03-26 16:19:24.853302+0500 TestApp[14012:243106] ℹ️-[TSLocationManager doStart:] trackingMode: 1
2021-03-26 16:19:24.854055+0500 TestApp[14012:243106] 🎾-[TSGeofenceManager start]
2021-03-26 16:19:24.855035+0500 TestApp[14012:243129] [javascript] '[providerchange] -', false, 0
2021-03-26 16:19:24.859006+0500 TestApp[14012:243138] 🔵-[TSLocationManager setPace:] 0
2021-03-26 16:19:24.859312+0500 TestApp[14012:242742] [Client] {"msg":"#NullIsland Either the latitude or longitude was exactly 0! That's highly unlikely", "latIsZero":0, "lonIsZero":0}
2021-03-26 16:19:24.871794+0500 TestApp[14012:242742] [Client] {"msg":"#NullIsland Received a latitude or longitude from getLocationForBundleID that was exactly zero", "latIsZero":0, "lonIsZero":0, "location":'18 AB F0 39 A2 7F 00 00'}
2021-03-26 16:19:24.874335+0500 TestApp[14012:243138] 🎾-[TSLocationManager startUpdatingLocation] Location-services: ON
2021-03-26 16:19:24.888331+0500 TestApp[14012:243129] [javascript] '- BackgroundGeolocation is configured and ready: ', true
2021-03-26 16:19:25.281281+0500 TestApp[14012:243106]
╔═══════════════════════════════════════════════════════════
║ -[TSHttpService flush:]
╚═══════════════════════════════════════════════════════════
2021-03-26 16:19:25.283052+0500 TestApp[14012:243097] ✅-[LocationDAO unlock]_block_invoke UNLOCKED ALL RECORDS
2021-03-26 16:19:25.284148+0500 TestApp[14012:243097]
╔═══════════════════════════════════════════════════════════
║ -[TSHttpService finish:error:] Success: 1
╚═══════════════════════════════════════════════════════════
2021-03-26 16:19:25.292087+0500 TestApp[14012:243097] ℹ️+[LocationAuthorization run:onCancel:] status: 0
2021-03-26 16:19:25.296784+0500 TestApp[14012:242742] -[TSBackgroundFetch addListener:callback:timeout:]: unrecognized selector sent to instance 0x600003ae6c00
2021-03-26 16:19:25.297375+0500 TestApp[14012:243097] 🔵+[LocationAuthorization run:onCancel:] Request: requestAlwaysAuthorization
2021-03-26 16:19:25.297915+0500 TestApp[14012:243097] 🎾-[TSLocationManager startMonitoringBackgroundFetch] BackgroundFetch: ON
2021-03-26 16:19:25.305010+0500 TestApp[14012:242742] *** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: '-[TSBackgroundFetch addListener:callback:timeout:]: unrecognized selector sent to instance 0x600003ae6c00'
*** First throw call stack:
(
0 CoreFoundation 0x00007fff20421af6 __exceptionPreprocess + 242
1 libobjc.A.dylib 0x00007fff20177e78 objc_exception_throw + 48
2 CoreFoundation 0x00007fff204306f7 +[NSObject(NSObject) instanceMethodSignatureForSelector:] + 0
3 CoreFoundation 0x00007fff20426036 __forwarding__ + 1489
4 CoreFoundation 0x00007fff20428068 _CF_forwarding_prep_0 + 120
5 TestApp 0x000000010caf33c2 -[TSLocationManager startMonitoringBackgroundFetch] + 383
6 libdispatch.dylib 0x000000010fd8a7ec _dispatch_call_block_and_release + 12
7 libdispatch.dylib 0x000000010fd8b9c8 _dispatch_client_callout + 8
8 libdispatch.dylib 0x000000010fd99e75 _dispatch_main_queue_callback_4CF + 1152
9 CoreFoundation 0x00007fff2038fdbb _CFRUNLOOP_IS_SERVICING_THE_MAIN_DISPATCH_QUEUE_ + 9
10 CoreFoundation 0x00007fff2038a63e __CFRunLoopRun + 2685
11 CoreFoundation 0x00007fff203896d6 CFRunLoopRunSpecific + 567
12 GraphicsServices 0x00007fff2c257db3 GSEventRunModal + 139
13 UIKitCore 0x00007fff24696cf7 -[UIApplication _run] + 912
14 UIKitCore 0x00007fff2469bba8 UIApplicationMain + 101
15 TestApp 0x000000010c232a60 main + 112
16 libdyld.dylib 0x00007fff2025a3e9 start + 1
17 ??? 0x0000000000000001 0x0 + 1
)
libc++abi.dylib: terminating with uncaught exception of type NSException
*** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: '-[TSBackgroundFetch addListener:callback:timeout:]: unrecognized selector sent to instance 0x600003ae6c00'
terminating with uncaught exception of type NSException
CoreSimulator 732.18.6 - Device: iPhone 11 Pro (DA706599-DD81-4D09-8B19-DB4004D8C2C8) - Runtime: iOS 14.4 (18D46) - DeviceType: iPhone 11 Pro
(lldb)

**Comment by mujeebsardar** [comment]
sir this StackTrack is similar to this issue kindly review i am truble @christocracy

**Comment by mujeebsardar** [comment]
DirectionMaps driction', 74.36278
2021-03-26 16:22:38.710761+0500 TestApp[14073:246026] Metal API Validation Enabled
2021-03-26 16:22:38.797228+0500 TestApp[14073:246146] ℹ️-[TSLocationManager on:success:failure:] location
2021-03-26 16:22:38.797643+0500 TestApp[14073:246146] ℹ️-[TSLocationManager on:success:failure:] motionchange
2021-03-26 16:22:38.798043+0500 TestApp[14073:246146] ℹ️-[TSLocationManager on:success:failure:] activitychange
2021-03-26 16:22:38.798235+0500 TestApp[14073:246146] ℹ️-[TSLocationManager on:success:failure:] providerchange
2021-03-26 16:22:38.896691+0500 TestApp[14073:246135] ℹ️-[TSConfig persist]
2021-03-26 16:22:38.901204+0500 TestApp[14073:246144] ℹ️-[TSConfig persist]
2021-03-26 16:22:38.904214+0500 TestApp[14073:246158] 🔵-[TSLocationManager ready]
2021-03-26 16:22:38.906587+0500 TestApp[14073:246158] ℹ️-[TSConfig didDeviceReboot] didDeviceReboot? 0
2021-03-26 16:22:38.907037+0500 TestApp[14073:246158] ℹ️-[TSLocationManager doStart:] trackingMode: 1
2021-03-26 16:22:38.907097+0500 TestApp[14073:246155] [javascript] '[providerchange] -', true, 4
2021-03-26 16:22:38.908077+0500 TestApp[14073:246158] 🎾-[TSGeofenceManager start]
2021-03-26 16:22:38.913915+0500 TestApp[14073:246144] 🔵-[TSLocationManager setPace:] 0
2021-03-26 16:22:38.914408+0500 TestApp[14073:246026] [Client] {"msg":"#NullIsland Either the latitude or longitude was exactly 0! That's highly unlikely", "latIsZero":0, "lonIsZero":0}
2021-03-26 16:22:38.923916+0500 TestApp[14073:246144] 🎾-[TSLocationManager startUpdatingLocation] Location-services: ON
2021-03-26 16:22:38.928793+0500 TestApp[14073:246155] [javascript] '[location] -', { coords:
{ altitude: 0,
latitude: 31.4537,
heading: -1,
altitude_accuracy: -1,
heading_accuracy: -1,
accuracy: 5,
speed_accuracy: -1,
longitude: 74.299,
speed: -1,
floor: null },
timestamp: '2021-03-26T11:22:38.919Z',
extras: {},
sample: true,
is_moving: false,
uuid: 'E0C7CF43-675C-4B5F-B746-86F709191629',
odometer: 0,
activity: { confidence: 100, type: 'unknown' },
battery: { is_charging: false, level: -1 } }
2021-03-26 16:22:39.017019+0500 TestApp[14073:246155] [javascript] '- BackgroundGeolocation is configured and ready: ', true
2021-03-26 16:22:39.126328+0500 TestApp[14073:246135]
📍<+31.45370000,+74.29900000> +/- 5.00m (speed -1.00 mps / course -1.00) @ 3/26/21, 4:22:37 PM Pakistan Standard Time
2021-03-26 16:22:39.126883+0500 TestApp[14073:246135]
╔═══════════════════════════════════════════════════════════
║ -[TSLocationManager locationManager:didUpdateLocations:] Enabled: 1 | isMoving: 0 | df: -1.0m | age: 1.2s
╚═══════════════════════════════════════════════════════════
2021-03-26 16:22:39.127293+0500 TestApp[14073:246135] ✅-[TSLocationManager locationManager:didUpdateLocations:] Acquired motionchange position: <+31.45370000,+74.29900000> +/- 5.00m (speed -1.00 mps / course -1.00) @ 3/26/21, 4:22:37 PM Pakistan Standard Time
2021-03-26 16:22:39.128726+0500 TestApp[14073:246135] 🔵-[TSLocationManager startMonitoringStationaryRegion:radius:] Radius: 25
2021-03-26 16:22:39.133550+0500 TestApp[14073:246158] 🔴-[TSLocationManager stopUpdatingLocation]
2021-03-26 16:22:39.138028+0500 TestApp[14073:246158] 🔵-[TSLocationManager calculateMedianLocationAccuracy:] Median location accuracy: 5.0
2021-03-26 16:22:39.138762+0500 TestApp[14073:246155] [javascript] '[motionchange] -', false, { battery: { level: -1, is_charging: false },
activity: { type: 'unknown', confidence: 100 },
event: 'motionchange',
odometer: 0,
uuid: 'CF7F94B0-37BA-4B12-83EA-DFF095EEEA39',
is_moving: false,
extras: {},
timestamp: '2021-03-26T11:22:37.942Z',
coords:
{ floor: null,
speed: -1,
longitude: 74.299,
speed_accuracy: -1,
accuracy: 5,
heading_accuracy: -1,
altitude_accuracy: -1,
heading: -1,
latitude: 31.4537,
altitude: 0 } }
2021-03-26 16:22:39.142456+0500 TestApp[14073:246155] [javascript] '[location] -', { coords:
{ altitude: 0,
latitude: 31.4537,
heading: -1,
altitude_accuracy: -1,
heading_accuracy: -1,
accuracy: 5,
speed_accuracy: -1,
longitude: 74.299,
speed: -1,
floor: null },
timestamp: '2021-03-26T11:22:37.942Z',
extras: {},
is_moving: false,
uuid: 'CF7F94B0-37BA-4B12-83EA-DFF095EEEA39',
odometer: 0,
event: 'motionchange',
activity: { confidence: 100, type: 'unknown' },
battery: { is_charging: false, level: -1 } }
2021-03-26 16:22:39.176762+0500 TestApp[14073:246026] -[TSBackgroundFetch addListener:callback:timeout:]: unrecognized selector sent to instance 0x6000009fa4c0
2021-03-26 16:22:39.146111+0500 TestApp[14073:246158] 🎾-[TSLocationManager startMonitoringSignificantLocationChanges]
2021-03-26 16:22:39.163869+0500 TestApp[14073:246146] App Transport Security has blocked a cleartext HTTP (http://) resource load since it is insecure. Temporary exceptions can be configured via your app's Info.plist file.
2021-03-26 16:22:39.179920+0500 TestApp[14073:246158] ✅-[LocationDAO unlock]_block_invoke UNLOCKED ALL RECORDS
2021-03-26 16:22:39.180067+0500 TestApp[14073:246158] ✅-[TSLocationManager persistLocation:]_block_invoke INSERT: CF7F94B0-37BA-4B12-83EA-DFF095EEEA39
2021-03-26 16:22:39.180030+0500 TestApp[14073:246146] Cannot start load of Task <3F5BA86F-59AE-4C77-9129-910FF15600F9>.<1> since it does not conform to ATS policy
2021-03-26 16:22:39.180221+0500 TestApp[14073:246158]
╔═══════════════════════════════════════════════════════════
║ -[TSHttpService flush:]
╚═══════════════════════════════════════════════════════════
2021-03-26 16:22:39.180338+0500 TestApp[14073:246158] ✅-[BackgroundTaskManager createBackgroundTask] 3
2021-03-26 16:22:39.181636+0500 TestApp[14073:246158] ✅-[TSHttpService schedulePost] LOCKED: CF7F94B0-37BA-4B12-83EA-DFF095EEEA39
2021-03-26 16:22:39.186131+0500 TestApp[14073:246026] *** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: '-[TSBackgroundFetch addListener:callback:timeout:]: unrecognized selector sent to instance 0x6000009fa4c0'
*** First throw call stack:
(
0 CoreFoundation 0x00007fff20421af6 __exceptionPreprocess + 242
1 libobjc.A.dylib 0x00007fff20177e78 objc_exception_throw + 48
2 CoreFoundation 0x00007fff204306f7 +[NSObject(NSObject) instanceMethodSignatureForSelector:] + 0
3 CoreFoundation 0x00007fff20426036 __forwarding__ + 1489
4 CoreFoundation 0x00007fff20428068 _CF_forwarding_prep_0 + 120
5 TestApp 0x0000000100fab3c2 -[TSLocationManager startMonitoringBackgroundFetch] + 383
6 libdispatch.dylib 0x00000001042427ec dispatch_call2021-03-26 16:22:39.183618+0500 TestApp[14073:246143] Task <3F5BA86F-59AE-4C77-9129-910FF15600F9>.<1> finished with error [-1022] Error Domain=NSURLErrorDomain Code=-1022 "The resource could not be loaded because the App Transport Security policy requires the use of a secure connection." UserInfo={NSLocalizedDescription=The resource could not be loaded because the App Transport Security policy requires the use of a secure connection., NSErrorFailingURLStringKey=http://yourserver.com/locations, NSErrorFailingURLKey=http://yourserver.com/locations, NSURLErrorRelatedURLSessionTaskErrorKey=(
"LocalDataTask <3F5BA86F-59AE-4C77-9129-910FF15600F9>.<1>"
), _NSURLErrorFailingURLSessionTaskErrorKey=LocalDataTask <3F5BA86F-59AE-4C77-9129-910FF15600F9>.<1>, NSUnderlyingError=0x6000012be4c0 {Error Domain=kCFErrorDomainCFNetwork Code=-1022 "(null)"}}
_block_and_release + 12
7 libdispatch.dylib 0x00000001042439c8 _dispatch_client_callout + 8
8 libdispatch.dylib 0x0000000104251e75 _dispatch_main_queue_callback_4CF + 1152
9 CoreFoundation 0x00007fff2038fdbb _CFRUNLOOP_IS_SERVICING_THE_MAIN_DISPATCH_QUEUE_ + 9
10 CoreFoundation 0x00007fff2038a63e __CFRunLoopRun + 2685
11 CoreFoundation 0x00007fff203896d6 CFRunLoopRunSpecific + 567
12 GraphicsServices 0x00007fff2c257db3 GSEventRunModal + 139
13 UIKitCore 0x00007fff24696cf7 -[UIApplication _run] + 912
14 UIKitCore 0x00007fff2469bba8 UIApplicationMain + 101
15 TestApp 0x00000001006eaa60 main + 112
16 libdyld.dylib 0x00007fff2025a3e9 start + 1
17 ??? 0x0000000000000001 0x0 + 1
)
libc++abi.dylib: terminating with uncaught exception of type NSException
2021-03-26 16:22:39.188473+0500 TestApp[14073:246158] ℹ️+[LocationAuthorization run:onCancel:] status: 4
2021-03-26 16:22:39.189216+0500 TestApp[14073:246158] ℹ️-[TSConfig persist]
2021-03-26 16:22:39.202580+0500 TestApp[14073:246158] 🎾-[TSLocationManager startMonitoringBackgroundFetch] BackgroundFetch: ON
2021-03-26 16:22:39.203041+0500 TestApp[14073:246158] ⚠️-[HttpResponse handleResponse] HTTP ERROR: 0
***
* The resource could not be loaded because the App Transport Security policy requires the use of a secure connection.
*
{
NSErrorFailingURLKey = "http://yourserver.com/locations";
NSErrorFailingURLStringKey = "http://yourserver.com/locations";
NSLocalizedDescription = "The resource could not be loaded because the App Transport Security policy requires the use of a secure connection.";
NSUnderlyingError = "Error Domain=kCFErrorDomainCFNetwork Code=-1022 \"(null)\"";
"_NSURLErrorFailingURLSessionTaskErrorKey" = "LocalDataTask <3F5BA86F-59AE-4C77-9129-910FF15600F9>.<1>";
"_NSURLErrorRelatedURLSessionTaskErrorKey" = (
"LocalDataTask <3F5BA86F-59AE-4C77-9129-910FF15600F9>.<1>"
);
}
2021-03-26 16:22:39.204171+0500 TestApp[14073:246158] ✅-[TSHttpService post:]_block_invoke UNLOCKED: CF7F94B0-37BA-4B12-83EA-DFF095EEEA39
2021-03-26 16:22:39.204992+0500 TestApp[14073:246158]
╔═══════════════════════════════════════════════════════════
║ -[TSHttpService finish:error:] Success: 0
╚═══════════════════════════════════════════════════════════
